src/pages/app/index.ts

```diff
@@ -0,0 +1,72 @@
+import Page from '../../types/page';
+import MainPage from '../main';
+import CatalogPage from '../catalog';
+import ProductPage from '../product';
+import CartPage from '../cart';
+import ErrorPage, { ErrorTypes } from '../error';
+import Header from '../../components/header';
+import { Footer } from '../../components/footer';
+import { routes } from '../../enums';
+
+class App {
+    private static container: HTMLElement = document.body;
+    private static defaultPageId = 'current-page';
+    private header: Header;
+    private footer: Footer;
+    private mainWrapper: HTMLElement;
+
+    constructor() {
+        this.header = new Header('header', 'header');
+        this.footer = new Footer('footer', 'footer');
+        this.mainWrapper = document.createElement('main');
+    }
+
+    renderNewPage(idPage: string) {
+        const currentPageHTML = document.querySelector(`#${App.defaultPageId}`);
+
+        if (currentPageHTML) {
+            currentPageHTML.remove();
+        }
+
+        let page: Page | null = null;
+
+        if (idPage === routes.MainPage) {
+            page = new MainPage(idPage);
+        } else if (idPage.includes(routes.CatalogPage)) {
+            page = new CatalogPage(idPage);
+        } else if (idPage === routes.CartPage) {
+            page = new CartPage(idPage);
+        } else if (idPage.includes(routes.ProductPage)) {
+            page = new ProductPage(idPage);
+        } else {
+            page = new ErrorPage(idPage, ErrorTypes.NotFound);
+        }
```

Comment:
Please, try to avoid `let`
It will be better to use separate func: `const page = getPage(idPage)`